packages/core/useAsyncQueue/index.ts

```diff
@@ -0,0 +1,96 @@
+import { ref, Ref } from 'vue-demi'
+import { noop } from '@vueuse/shared'
+
+interface Result {
+  state: 'pending' | 'fulfilled' | 'rejected'
+  data: any
+}
```

Comment:
```suggestion
interface UseAsyncQueueResult<T> {
  state: 'pending' | 'fulfilled' | 'rejected'
  data: T
}
```

We'd better support generic types